### PR TITLE
Add FXIOS-6662 [v117] Handle ETP coordinator transitioning Delegate

### PR DESCRIPTION
**Pull request by aalaaeid**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6662)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14890)

## :bulb: Description

1. this make the ETPCoordinator the transitioning delegate.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and ensured the tests suite is passing
- [ ] Implemented accessibility and tested on UI related work (minimum Dynamic Text and VoiceOver)
- [ ] Updated documentation / comments for complex code and public methods if needed

### CODE REVIEW

File: Client/Coordinators/BaseCoordinator.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import Common
 import Foundation
 
-open class BaseCoordinator: Coordinator {
+open class BaseCoordinator: NSObject, Coordinator {
```
Comment (lmarceau) [inline Client/Coordinators/BaseCoordinator.swift:8]: Why was `NSObject` needed?

File: Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift [hunk 2]
```diff
@@ -43,7 +43,15 @@ class EnhancedTrackingProtectionCoordinator: BaseCoordinator,
     }
 
     func start() {
-        router.push(enhancedTrackingProtectionMenuVC)
+        enhancedTrackingProtectionMenuVC.modalPresentationStyle = .custom
```
Comment (lmarceau) [inline Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:46]: In the legacy code we had"
```
        if UIDevice.current.userInterfaceIdiom == .phone {
            etpVC.modalPresentationStyle = .custom
```
So I think in the case of iPad we still want to use the popover instead of the custom one.

File: Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift [hunk 3]
```diff
@@ -43,7 +43,15 @@ class EnhancedTrackingProtectionCoordinator: BaseCoordinator,
...
+        enhancedTrackingProtectionMenuVC.transitioningDelegate = self
+        let keyWindow = UIApplication.shared.windows.filter {$0.isKeyWindow}
```
Comment (lmarceau) [inline Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:48]: nit
```suggestion
        let keyWindow = UIApplication.shared.windows.filter { $0.isKeyWindow }
```

File: Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift [hunk 4]
```diff
@@ -43,7 +43,15 @@ class EnhancedTrackingProtectionCoordinator: BaseCoordinator,
...
+        if var topController = keyWindow.first?.rootViewController {
+            while let presentedViewController = topController.presentedViewController {
+                topController = presentedViewController
+            }
+            topController.present(enhancedTrackingProtectionMenuVC, animated: true, completion: nil)
+        }
```
Comment (lmarceau) [inline Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:54]: Why was this needed to replace `router.push(enhancedTrackingProtectionMenuVC)` ? We should keep pushing from the router with the coordinator

File: Client/Coordinators/BaseCoordinator.swift (hunk 1 above)
Comment (aalaaeid) [reply Client/Coordinators/BaseCoordinator.swift:8]: `NSObject` needed because of this error:

`/firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:68:1: Cannot declare conformance to 'NSObjectProtocol' in Swift; 'EnhancedTrackingProtectionCoordinator' should inherit 'NSObject' instead`

File: Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift (hunk 4 above)
Comment (aalaaeid) [reply Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:54]: @lmarceau  The issue is that the enhancedTrackingProtectionMenuVC is pushed with an empty controller if it is pushed from the last navigation controller, not the top controller.

File: Client/Coordinators/BaseCoordinator.swift (hunk 1 above)
Comment (lmarceau) [reply Client/Coordinators/BaseCoordinator.swift:8]: Ah ok when we conformed to `UIViewControllerTransitioningDelegate` 👍

File: Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift (hunk 4 above)
Comment (lmarceau) [reply Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:54]: Ah, I think it's because there's no `rootViewController` to this navigation controller. Can you try `router.setRootViewController(enhancedTrackingProtectionMenuVC)`? (like it was done for settings coordinator [here](https://github.com/mozilla-mobile/firefox-ios/blob/aa74c92e634a34e15db231bfee14df4a4d9ddff0/Client/Coordinators/SettingsCoordinator.swift#L42))

Comment (aalaaeid) [reply Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:54]: @lmarceau  Yes, I tried this, but it keeps crashing

**Comment by lmarceau** [comment]
Almost there, Bitrise failed with:
```
❌  /Users/vagrant/git/Tests/ClientTests/Coordinators/BrowserCoordinatorTests.swift:661:48: missing argument for parameter 'sourceView' in call
        subject.showEnhancedTrackingProtection()
```

**Review by lmarceau — APPROVED** [review]
